### ISSUE TITLE
OBS-pc spontaan uitgevallen om 02:16 (17-09) en niet vanzelf teruggekomen

### ISSUE THREAD

**Issue by peterdeswart96-ship-it** [peterdeswart96-ship-it/mokum-streams#133]
## Wat er aan de hand is
De OBS-pc is vannacht om **02:16** weggevallen en stond om 10:00 nog steeds uit.

Bewijs:
- Dashboard: "Agent offline — laatst gezien 26973s geleden" (backend-tijd 09:46 → ±02:15)
- Chrome Remote Desktop: `Mokum OBS — Last online: 2:16 AM`
- Backend-logs rond dat moment: niets over de pc. De enige regel in de buurt (`WorkerTerminate`, 02:34) is een Azure-interne worker-recycle.

## Wat het NIET is
- **Niet de wekelijkse herstart.** Die staat op ma/do **06:00** (taak `MokumWeeklyRestart`, zie `docs/obs-pc-autostart.md`). De pc viel bijna vier uur te vroeg weg.
- **Niet al eerder offline.** De pc heeft de avond van 16-09 gewoon gedraaid: tafel 1 streamde tot 01:06 (5:45:41, netjes gefinaliseerd met 10 hoofdstukken) en de agent bleef daarna melden tot ±02:15.
- **Niet Windows Update.** Actieve uren staan op 12:00–06:00 en `NoAutoRebootWithLoggedOnUsers=1`; 02:16 valt binnen het beschermde venster.

Dit lijkt op de signatuur van het incident van **12-09** (pc zelf loopt vast, niet OBS of de camera). Precies daarvoor is de wekelijkse herstart destijds ingesteld.

## Het gat in het vangnet
Een geplande herstart redt een pc niet die al dood is. De taak van 06:00 heeft vanochtend nooit kunnen draaien. Als bescherming tegen vastlopers werkt hij alleen zolang de pc het tot 06:00 volhoudt.

## Diagnose zodra de pc weer draait
```powershell
# Onverwacht uitgevallen (41) of netjes afgesloten (1074)?
Get-WinEvent -FilterHashtable @{LogName='System'; Id=41,1074,6008,6005} -MaxEvents 15 |
  Select-Object TimeCreated, Id, Message | Format-List

# Heeft de wekelijkse herstart ooit gedraaid?
Get-ScheduledTask -TaskName 'MokumWeeklyRestart' | Get-ScheduledTaskInfo |
  Select LastRunTime, LastTaskResult, NextRunTime

# Zijn OBS en de agent na de boot teruggekomen?
Get-ScheduledTask -TaskName 'MokumOBS-Autostart' | Get-ScheduledTaskInfo | Select LastRunTime, LastTaskResult
@(Get-Process obs64 -ErrorAction SilentlyContinue).Count   # moet 4 zijn
Get-ScheduledTask -TaskName 'MokumAgent' | Get-ScheduledTaskInfo | Select LastRunTime, LastTaskResult
```

## Te doen
- [ ] Windows-eventlog uitlezen zodra iemand bij de pc kan (±12:00)
- [ ] Openstaande punten uit #43 die hier direct op ingrijpen afmaken:
  - [ ] BIOS: restore on power loss = on (komt vanzelf terug na een harde uitval)
  - [ ] Fast Startup uit (een herstart wordt dan een echte, schone boot)
- [ ] Overwegen: hardware-watchdog of een slimme stekker, zodat een dode pc ook zonder handen terugkomt

## Hangt samen met
- #43 (OBS-pc 24/7 betrouwbaar maken)
- #60 (Wake-on-LAN)
- #132 (alarm als de agent offline gaat — had dit om 02:20 gemeld)

**Comment by peterdeswart96-ship-it** [comment]
## Eventlog uitgelezen (17-09, ±16:00)

**17-09: harde uitval, geen nette afsluiting.**
- Event 6008: "vorige afsluiting om **01:41:43** onverwacht". Dat is het laatste 'levensteken' dat Windows periodiek wegschrijft. De agent meldde zich nog tot ±02:15, dus de pc is tussen 01:41 en 02:16 uitgevallen.
- Event 41 om 15:10:35: opnieuw gestart zonder correct afsluiten. Er is geen 1074, dus geen herstart door Windows, een taak of een mens.
- Pas om **15:10** weer aan. Er kwam dus iemand langs: de pc gaat na uitval niet vanzelf aan, want "restore on power loss" in de BIOS staat nog open (#43).

**15-09: nóg een harde uitval, die nog niet bekend was.**
- 6008: onverwacht uit om **11:21:34** (overdag, zonder stream), weer aan om 15:05.
- Daarna drie herstarts door Windows Update (TrustedInstaller, 15:05–15:07) en een handmatige herstart via het startmenu om 16:20.

**Patroon:** 12-09, 15-09 en 17-09 is **drie keer vastlopen of uitvallen in zes dagen**. Eén keer gebeurde dat overdag zonder belasting. Dat wijst eerder op hardware, voeding of een driver dan op OBS of de streams. De wekelijkse herstart houdt dit niet tegen.

**Taken na de boot:**
- `MokumOBS-Autostart` → 0 (ok), 4× obs64 actief.
- `MokumAgent` → 267009 = 0x41301 "draait nog" (ok).
- `MokumWeeklyRestart` → LastRunTime **17-09 15:13:52**, resultaat 2147946720 = **0x800710E0** ("verzoek geweigerd"). Hij draaide dus 3 minuten na de boot, buiten ma/do 06:00, en herstartte niet (geen 1074). **Nog niet verklaard**, zie de vervolgcheck.

## Vervolgdiagnose
- Event 41 → `BugcheckCode`: 0 = bevriezing of stroomverlies, iets anders = blauw scherm (dan staat er een dump).
- Hardwarefouten (WHEA-Logger), fouten van de NVIDIA-driver (nvlddmkm) en minidumps.
- Instellingen van `MokumWeeklyRestart` (`StartWhenAvailable`?). Draait hij na elke boot een gemiste herstart in, dan kan de pc op een streamavond 3 minuten na het opstarten herstarten.

**Comment by peterdeswart96-ship-it** [comment]
## Vervolgdiagnose (17-09)

| Check | Uitkomst | Betekenis |
|---|---|---|
| Event 41 `BugcheckCode` | **0** bij 24-08 12:38, 15-09 15:05 en 17-09 15:10 | Geen blauw scherm |
| Event 41 `PowerButtonTimestamp` | **0** bij alle drie | Niemand heeft een draaiende pc met de aan/uit-knop uitgezet |
| `C:\Windows\Minidump` | leeg | Past bij "geen blauw scherm" |
| WHEA / nvlddmkm / disk / Display sinds 10-09 | niets (alleen de 41's) | Windows zag geen CPU-, geheugen-, GPU- of schijffout aankomen |
| `MokumWeeklyRestart` `StartWhenAvailable` | **False** | Een gemiste herstart wordt niet ingehaald. De run van 15:13 is daarmee niet verklaard, maar ook niet gevaarlijk |

Er is ook een derde harde uitval die nog niet bekend was: **24-08, 12:38**.

## Wat er overblijft
Twee verklaringen passen:
1. **De stroom viel weg** (stekkerdoos uit, groep uit, stekker eruit, voeding van de pc zelf).
2. **Volledige bevriezing** zonder logregel (oververhitting, voeding die even inzakt), waarna de stroom er alsnog af ging.

Het verschil zit in één vraag aan wie de pc om 15:10 aanzette: **was hij helemaal uit** (geen lampjes, geen ventilatoren), of **stond hij aan met een bevroren of zwart scherm**?

De tijdstippen 01:41 (na sluiting) en 11:21 (ochtend) passen bij iemand die bij sluiten of openen een stekkerdoos of groep omzet. Dat is nog niet bewezen.

## Voorstel
- **BIOS: restore on AC power loss = Power On** (openstaand uit #43) krijgt nu voorrang. Bij verklaring 1 komt de pc dan vanzelf terug, en autologin + `MokumOBS-Autostart` doen de rest.
- Nagaan waar de stekker van de pc in zit en wat daar nog meer aan hangt of mee uitgeschakeld wordt.
- Overwegen: een kleine UPS. Die overbrugt stroomdips en laat zien of het stroom was.
